### PR TITLE
Fixed Invite SDK  

### PR DESCRIPTION
**Pull request by danwalkowski**
It now correctly builds and runs, and should be less brittle in the future.
I updated the project files, fixed some outdated calls, and eliminated a leak.

Everything seems to work

### CODE REVIEW

**Comment by danwalkowski** [comment]
I'm giving myself a thumbs up: +1+

**Comment by shortstuffsushi** [comment]
Assuming you had the conversation internally to drop 6.0. At this point, probably fair, but worth noting here for others who find out the hard way.

File: Classes/Invite/Providers/AddressBook/BranchInviteAddressBookContact.m [hunk 1]
```diff
@@ -11,7 +11,6 @@
 @implementation BranchInviteAddressBookContact
 
 - (void)dealloc {
-    CFRelease(self.contact);
```
Comment (shortstuffsushi) [inline Classes/Invite/Providers/AddressBook/BranchInviteAddressBookContact.m]: Why is this removed? And if so, the whole method impl should be deleted.

File: Classes/Invite/Providers/AddressBook/BranchInviteTextContactProvider.m [hunk 2]
```diff
@@ -123,6 +123,7 @@ - (void)loadContactsFromAddressBookIfPossible:(callbackWithStatus)callback {
         [self loadContactsFromAddressBook:addressBook callback:callback];
     }
     else {
+        CFRelease(addressBook);
```
Comment (shortstuffsushi) [inline Classes/Invite/Providers/AddressBook/BranchInviteTextContactProvider.m]: Why is this only added here? Is it eventually released elsewhere?

Comment (danwalkowski) [inline Classes/Invite/Providers/AddressBook/BranchInviteTextContactProvider.m]: Xcode complained it was leaking.  I added the CFRelease to solve that, but that created another issue, a timing issue with the use of the addressBook.  If you run the code normally, it crashes in the subsequent method that sorts the entries.  

Step through the code, and it works correctly, run it, and it crashes.  A race condition involving the addressBook.  I reverted the code, since a small leak is preferable to a crash.

In the meantime, I got halfway through submitting it (tags created, so I needed to skip a version)

So the net result is that I updated the dependencies, allowing the invite sdk to use the latest version of HMSegmentedController, and got it building with the latest Xcode.
